Code/Scripts/metrics.py

```diff
@@ -0,0 +1,257 @@
+        #print(i)
+        sum=sum+i
+    mean=(sum/(sample_num-1))
+    ME_wop = list(np.round(mean,2))
+    #print(ME_wop)
+    
+    ##MDE - Median Error
+    #for i in range(param_num):
+     #   param_i = []
+    #for row in param_diff:
+     #   for i in range(param_num):
+      #      param_i.append(row[i])
+    #for i in param_num: 
+     #   print(param_i)
+            
+            
+    #Median_wop = list(np.round(median(param_diff),2))
+
+     #WP
+    param_diff=[]
+    for i in range(sample_num-1):
+        param_diff.append(y_pred_wp[i]-y_test[i])
+    #print(param_diff)
+    ##ME - Mean Error
+    sum=0
+    for i in param_diff:
+        #print(i)
+        sum=sum+i
+    mean=(sum/(sample_num-1))
+    ME_wp = list(np.round(mean,2))
+    
+    #WPL
+    param_diff=[]
+    for i in range(sample_num-1):
+        param_diff.append(y_pred_wpl[i]-y_test[i])
+    ##ME - Mean Error
+    sum=0
+    for i in param_diff:
+        sum=sum+i
+    mean=(sum/(sample_num-1))
+    ME_wpl = list(np.round(mean,2))
+    #print(ME_wpl)
+    ##MDE - Median Error
+    #Median_wpl = list(np.round(median(param_diff),2))
+
+    ##Msigma - Mean Standard Deviation
+    sum=0
+    for i in std_pred_wop:
+        sum=sum+i
+    mean=(sum/(sample_num-1))
+    Mstd_wop = list(np.round(mean,2))
+    #print(Mstd_wop)
+    ##MDsigma - Median Standard Deviation
+    #MDstd_wop = list(np.round(median(std_pred_wop),2))
+
+    ##Msigma - Mean Standard Deviation
+    sum=0
+    for i in std_pred_wp:
+        sum=sum+i
+    mean=(sum/(sample_num-1))
+    Mstd_wp = list(np.round(mean,2))
+    #print(Mstd_wp)
+    ##MDsigma - Median Standard Deviation
+    #MDstd_wp = list(np.round(median(std_pred_wp),2))
+    
+    ##Msigma - Median Standard Deviation
+    sum=0
+    for i in std_pred_wpl:
+        sum=sum+i
+    mean=(sum/(sample_num-1))
+    Mstd_wpl = list(np.round(mean,2))
+    #print(Mstd_wpl)
+    ##MDsigma - Median Standard Deviation
+    #MDstd_wpl = list(np.round(median(std_pred_wpl),2))
+
+    mean_metrics = [ME_wop, ME_wp, ME_wpl, Mstd_wop, Mstd_wp, Mstd_wpl]
+    #median_metrics = [Median_wop, Median_wpl, MDstd_wop, MDstd_wpl]
+
+    return mean_metrics
+
+def PerturberSampleTrunc_Substructure(sample_num, param_num, y_test, y_pred_WoS, y_pred_WS, std_pred_WoS, std_pred_WS):
+
+    #WoS
+    param_diff=[]
+    for i in range(sample_num-1):
+        param_diff.append(y_pred_WoS[i]-y_test[i])
+    #print(param_diff)
+    
+    ##ME - Mean Error
+    sum=0
+    for i in param_diff:
+        sum=sum+i
+    mean=(sum/(sample_num-1))
+    ME_WoS = list(np.round(mean,2))
+    #print(ME_wop)
+
+     #WS
+    param_diff=[]
+    for i in range(sample_num-1):
+        param_diff.append(y_pred_WS[i]-y_test[i])
+    #print(param_diff)
+    
+    ##ME - Mean Error
+    sum=0
+    for i in param_diff:
+        sum=sum+i
+    mean=(sum/(sample_num-1))
+    ME_WS = list(np.round(mean,2))
+
+    ##Msigma - Mean Standard Deviation
+    sum=0
+    for i in std_pred_WoS:
```

Comment:
you should be able to call np.mean() here to do this concisely / with numpy built-ins which are the usually preferred